src/key.rs

```diff
@@ -323,4 +323,24 @@ mod tests {
 
         assert!(pk.verify(sig, msg));
     }
+
+    #[test]
+    fn test_from_bytes() {
+        // Larger than the modulus
+        assert!(PrivateKey::from_bytes(&[255u8; 32]).is_err());
+        // exactly the modulus
+        let modulus = Scalar::char_le_bits();
+
+        // save because little endian + size matches
+        let modulus_u8: &[u8] = unsafe { std::mem::transmute(modulus.as_raw_slice()) };
```

Comment:
If this `transmute` works, then this is ok with me. However, if you didn't want to use `unsafe` here to get the modulus, something which I've done in the past is hardcode the modulus' repr and test using:
```rust
// Scalar field modulus' bigint (i.e. non-Montgomery form) little-endian bytes.
let modulus_repr: [u8; 32] = [
    0x01, 0x00, 0x00, 0x00, 0xff, 0xff, 0xff, 0xff, 0xfe, 0x5b, 0xfe, 0xff, 0x02, 0xa4, 0xbd, 0x53,
    0x05, 0xd8, 0xa1, 0x09, 0x08, 0xd8, 0x39, 0x33, 0x48, 0x7d, 0x9d, 0x29, 0x53, 0xa7, 0xed, 0x73,
];

// Largest field element.
let neg1_repr: [u8; 32] = {
    let mut repr = modulus_repr;
    *repr[31] -= 1;
    repr
};

// Smallest integer greater than the modulus.
let modulus_plus_1_repr = {
    let mut repr = modulus_repr;
    *repr[31] += 1;
    repr
};

assert!(PrivateKey::from_bytes(&modulus_repr).is_err());
assert!(PrivateKey::from_bytes(&neg1_repr).is_ok());
assert!(PrivateKey::from_bytes(&modulus_plus_1_repr).is_err());
```